server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -2115,7 +2116,12 @@ public final EngineConfig getEngineConfig() {
     /**
      * Allows registering a listener for when the index shard is on a segment generation >= minGeneration.
      */
+    @Deprecated
```

Comment:
Can we document why it is deprecated and that `addPrimaryTermAndGenerationListener` should be used instead?